daemon/graphdriver/btrfs/btrfs.go

```diff
@@ -633,6 +634,10 @@ func (d *Driver) Remove(id string) error {
 	d.updateQuotaStatus()
 
 	if err := subvolDelete(d.subvolumesDir(), id, d.quotaEnabled); err != nil {
+		if sys.RunningInUserNS() && strings.Contains(err.Error(), "operation not permitted") {
+			// https://github.com/moby/moby/issues/41762
+			err = errors.Wrap(err, "failed to delete subvolume without root (hint: remount btrfs on \"user_subvol_rm_allowed\" option, or change the storage driver to \"fuse-overlayfs\")")
+		}
```

Comment:
Note: proper fix is being implemented in https://github.com/moby/moby/pull/42203 , but requires kernel >= 4.18.
So I think we can have both PRs.